cmd/templ/generatecmd/main.go

```diff
@@ -348,8 +447,26 @@ func generate(ctx context.Context, basePath, fileName string, generateSourceMapV
 		return nil, fmt.Errorf("%s source formatting error: %w", fileName, err)
 	}
 
-	if err = os.WriteFile(targetFileName, data, 0644); err != nil {
-		return nil, fmt.Errorf("%s write file error: %w", targetFileName, err)
+	// Hash and write the file if the hash has changed
+	hash := md5.Sum(data)
```

Comment:
`data` is not a good variable name. Not an issue in this PR, but I'd prefer it the name was `formattedGoCode`.